### PR TITLE
Modelnet

### PR DESCRIPTION
**Pull request by nicolas-chaulet**
I had to create a new version of ModelNet, the one on PYG is not very good for point clouds. People seem to use that one instead. Sampled meshes are very clean and regular.

### CODE REVIEW

File: torch_points3d/datasets/classification/modelnet.py [hunk 1]
```diff
@@ -1,12 +1,113 @@
...
+
+    @property
+    def processed_file_names(self):
+        return ["training_{}.pt".format(self.name), "test_{}.pt".format(self.name)]
+
+    def download(self):
+        path = download_url(self.url, self.root)
+        extract_zip(path, self.root)
+        os.unlink(path)
+        folder = osp.join(self.root, "modelnet40_normal_resampled")
```
Comment (humanpose1) [inline torch_points3d/datasets/classification/modelnet.py:74]: modelnet10 or modelnet40 right ?

File: conf/data/classification/modelnet.yaml [hunk 2]
```diff
@@ -6,6 +6,8 @@ data:
     number: 10
     pre_transforms:
         - transform: NormalizeScale
+        - transform: GridSampling
+            lparams: [0.02]
```
Comment (humanpose1) [inline conf/data/classification/modelnet.yaml:10]: Is a voxel size of 0.02 adapted for ModelNet  ?

File: torch_points3d/datasets/classification/modelnet.py (hunk 1 above)
Comment (nicolas-chaulet) [reply torch_points3d/datasets/classification/modelnet.py:74]: The zip file is the same, the distinction happens when creating the dataset

File: conf/data/classification/modelnet.yaml (hunk 2 above)
Comment (nicolas-chaulet) [reply conf/data/classification/modelnet.yaml:10]: Yes, it is in normalized coordinates (happens after the `NormalizeScale`). I am putting a notebook together to show all this. And it runs on Colab :)
https://colab.research.google.com/github/nicolas-chaulet/torch-points3d/blob/modelnettuto/dashboard/ObjectClassificationRSConv.ipynb

Comment (nicolas-chaulet) [reply conf/data/classification/modelnet.yaml:10]: image

**Comment by nicolas-chaulet** [comment]
The dataset if of better quality (at least visually) and contains normal vectors, so less risk of missing something and better quality

File: torch_points3d/datasets/base_dataset.py [hunk 3]
```diff
@@ -20,6 +20,25 @@
 log = logging.getLogger(__name__)
 
 
+def explode_transform(transforms):
```
Comment (tchaton) [inline torch_points3d/datasets/base_dataset.py:23]: We might want this one to be recursive no ? as a compose can have nested compose no ?

**Pull request merged**